hbase-balancer/src/main/java/org/apache/hadoop/hbase/master/balancer/BalancerConditionals.java

```diff
@@ -145,7 +145,7 @@ int getViolationCountChange(BalancerClusterState cluster, BalanceAction action)
     // Reset cluster
     cluster.doAction(undoAction);
 
-    if (isViolatingPre && isViolatingPost) {
+    if (isViolatingPre == isViolatingPost) {
       return 0;
```

Comment:
We also want to return 0 if we had zero violations both pre & post. Prior to this fix, we'd fall through to return -1 which erroneously considered a neutral move to be an improvement

Reply:
Since this was so impactful to behavior, maybe good to capture this method in a unit test?